docs/extensions/tutorials/_posts/1970-01-05-Installing3rdPartyPackages.md

```diff
@@ -0,0 +1,330 @@
+
+The reason we create an array is because sometimes our extension will have multiple native dependencies, so instead of making a separate linking command for each, we simply run one `forEach` loop and hand it an array of dependencies.
+
+### Using the Package
+
+Our extension will now have access to everything `react-native-camera` has to offer. For this example we'll edit `app/screens/QRReaderScreen.js` so that it displays an [alert](https://facebook.github.io/react-native/docs/alert.html) when the camera reads a QR code and the alert message will contain the QR code data.
+
+```javascript{2-4,6-10,12-21,25-28}
+#file: app/screens/QRReaderScreen.js
+import React, { Component } from 'react';
+import { Alert } from 'react-native';
+import Camera from 'react-native-camera';
+import _ from 'lodash';
+
+export default class QRReaderScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.onBarCodeRead = this.onBarCodeRead.bind(this);
+  }
+
+  onBarCodeRead(code) {       //when the bar (QR) code is read
+    Alert.alert(              //invoke this callback
+      'QR Code Detected',
+      code.data,
+      [
+        { text: 'OK, read it.', onPress: () => console.log('User saw QR Code contents.') },
+      ],
+      { cancelable: false },
+    );
+  }
+
+  render() {
+    return (
+      <Camera
+        onBarCodeRead={_.debounce(this.onBarCodeRead, 1000, { leading: true, trailing: false })}
+        style={{ flex: 1 }}
+      />
+    );
+  }
+}
+```
+
+After making these changes, we can push the extension to Shoutem and install it in a new app:
+
+```ShellSession
+$ shoutem push
+Uploading `QR Reader` extension to Shoutem...
+Success!
+$ shoutem install --new QRReader
+Extension installed
+See it in browser: `{{ site.shoutem.builderURL }}/app/{{ site.example.appId }}`
+```
+
+Opening the QRReader app in the Builder will show us an app with no Screens, but since we just installed our QR Reader Extension onto the app, we can just add the QRReader screen to it by clicking on the + button next to Screens, going to the Custom category and selecting the QR Reader Extension.
+
+This specific native dependency that we're using is already linked in the Shoutem Preview app, so we will be able to preview the app on our device. Click "Preview on Device" and scan the QR code using the Shoutem Preview app. Once it finishes building and loading our QRReader app, scan that same QR code and the app will produce an alert that contains our QRReader app's preview link.
+
+With any other native dependency, previewing the app through the Builder or through the Shoutem Preview app won't be possible, because we can't change the native code of the previewer, so instead we have to preview it [locally](http://shoutem.github.io/docs/extensions/tutorials/setting-local-environment) using `$ shoutem run-ios` or `$ shoutem run-android` and picking the QRReader app we just created.
```

Comment:
Remember to switch this to `react-native run-ios`/`run-android` once we'll have that available.